pkg/hive/internal/reflect.go

```diff
@@ -30,7 +31,25 @@ func FuncNameAndLocation(fn any) string {
 	name := TrimName(f.Name())
 	name = strings.TrimSuffix(name, "-fm")
 	if file != "<autogenerated>" {
-		return fmt.Sprintf("%s (%s:%d)", name, path.Base(file), line)
+		return fmt.Sprintf("%s (%s:%d)", name, usefulPathSegment(file), line)
 	}
 	return name
 }
+
+// Purely a heuristic.
+var commonRoots = map[string]struct{}{
+	"pkg": {},
+	"cmd": {},
+}
```

Comment:
What about also adding `api`?

Reply:
I don't think we have that many `api/*` objects in hive at the moment, do we? We can always add this later on.